pondo/pondo_core_protocol/src/main.leo

```diff
@@ -816,25 +1078,23 @@ program pondo_core_protocol.aleo {
     // At this point, all credits have been transferred to the core protocol, but there may still be commission owed
     let core_protocol_account: u64 = credits.aleo/account.get_or_use(self.address, 0u64);
     let reserved_for_withdrawal: u64 = balances.get(CLAIMABLE_WITHDRAWALS);
-    let deposit_pool: u64 = core_protocol_account - full_balance - reserved_for_withdrawal;
+    // Move bonded withdrawals to available to claim
+    balances.set(CLAIMABLE_WITHDRAWALS, reserved_for_withdrawal + bonded_withdrawals);
+    // Reset bonded withdrawals
+    balances.set(BONDED_WITHDRAWALS, 0u64);
+    // Calculate deposit pool
+    let deposit_pool: u64 = core_protocol_account - transfers_total - reserved_for_withdrawal;
 
     let new_commission_paleo: u64 = calculate_new_paleo(current_balance as u128, deposit_pool as u128, new_commission as u128, total_paleo_minted);
+    assert(commission_mint >= (current_owed_commission * 98u64) / 100u64); // Assert that the minted commission is at least 98% of the owed commission
     // New owed commission is whatever commission is left after the new commission mint, plus what we may have earned between calling the function and now
     owed_commission.set(0u8, current_owed_commission + new_commission_paleo - commission_mint);
     // Update total balance
     balances.set(DELEGATED_BALANCE, current_balance + new_commission);
 
-    // Move bonded withdrawals to available to claim
-    let current_epoch: u32 = block.height / BLOCKS_PER_EPOCH;
-    //  Process withdrawals from the previous epoch
-    let current_withdrawal_batch: u64 = withdrawal_batches.get_or_use(current_epoch - 1u32, 0u64);
-    balances.set(CLAIMABLE_WITHDRAWALS, reserved_for_withdrawal + current_withdrawal_batch);
-
-    // Update bonded withdrawals
-    let bonded_withdrawals: u64 = balances.get(BONDED_WITHDRAWALS);
-    balances.set(BONDED_WITHDRAWALS, bonded_withdrawals - current_withdrawal_batch);
-
```

Comment:
`Issue 05` Now we move all of `BONDED_WITHDRAWALS` to `CLAIMABLE` during a rebalance